app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/individual/FullIncomeSourceController.scala

```diff
@@ -125,6 +126,8 @@ class FullIncomeSourceController @Inject()(fullIncomeSource: FullIncomeSource,
   def backUrl(id: String, isEditMode: Boolean, isGlobalEdit: Boolean, isFirstBusiness: Boolean): String = {
     if (isEditMode || isGlobalEdit) {
       routes.SelfEmployedCYAController.show(id, isEditMode, isGlobalEdit).url
+    } else if (isFirstBusiness && isEnabled(RemoveAccountingMethod)) {
```

Comment:
```suggestion
    } else if (isEnabled(RemoveAccountingMethod)) {
```
I think you can remove that part, the new logic doesn't care about isFirstBusiness, so we can exclude that when the feature switch is enabled.